### PR TITLE
Refactoring of scripts and links for nonce propagation

### PR DESCRIPTION
**Pull request by eozmen410**

### CODE REVIEW

File: core/src/test/java/org/apache/struts2/views/jsp/ui/ComboBoxTest.java [hunk 1]
```diff
@@ -73,6 +73,8 @@ public void testSimple() throws Exception {
         tag.setId("cb");
         tag.setList("collection");
 
+        stack.getActionContext().getSession().put("nonce", "r4nd0m");
```
Comment (salcho) [inline core/src/test/java/org/apache/struts2/views/jsp/ui/ComboBoxTest.java:76]: why is this needed?

**Review by salcho — COMMENTED** [review]
thank you!

File: core/src/test/java/org/apache/struts2/views/jsp/ui/ComboBoxTest.java (hunk 1 above)
Comment (eozmen410) [reply core/src/test/java/org/apache/struts2/views/jsp/ui/ComboBoxTest.java:76]: This is putting the  nonce value in the session, just like the `CspInterceptor` does. We use it to test if the script blocks are rendered with nonce attributes.

**Review by salcho — APPROVED** [review]
wow this is great! looks much cleaner and easier to follow with the new changes. I wish some of these changes were on the original PR (moving the nonce to UIBean and so on) but I think we can live with it the way it is. Awesome work guys!

**Comment by salcho** [comment]
would you like me to merge this or would you prefer if we leave it here until we're ready for the new PR? @eozmen410 @gchatz22

**Comment by gchatz22** [comment]
I believe we are good to merge!

**Comment by eozmen410** [comment]
I think we can merge!
